src/controllers/FileDownloadController.ts

```diff
@@ -0,0 +1,38 @@
+import { Response } from 'express';
+import { OK } from 'http-status-codes';
+import { inject } from 'inversify';
+import { controller, httpGet, requestParam, response } from 'inversify-express-utils';
+import { BaseAsyncHttpController } from './BaseAsyncHttpController';
+
+import { loggerInstance } from 'app/middleware/Logger';
+import { FileTransferService } from 'app/service/FileTransferService';
+import { DOWNLOAD_FILE_URI } from 'app/utils/Paths';
+
+const template = 'download-file';
+
+@controller(DOWNLOAD_FILE_URI)
+export class FileDownloadController extends BaseAsyncHttpController {
+
+    constructor(@inject(FileTransferService) private readonly fileTransferService: FileTransferService) {
+        super();
+    }
+
+    @httpGet('/prompt/:fileId')
+    public renderPrompt(@response() res: Response, @requestParam('fileId') fileId: string): void {
+        res.render(template, { fileId });
+    }
+
+    @httpGet('/data/:fileId/download')
+    public async download(@response() res: Response, @requestParam('fileId') fileId: string): Promise<void> {
```

Comment:
is the response decorator needed?

Reply:
What do you actually mean by, is it needed?

Reply:
The same here I though about using `this.httpContext.response`.

Reply:
I can change it for consistency then